osu.Game.Tests/Visual/UserInterface/TestSceneMainMenuButton.cs

```diff
@@ -71,6 +72,7 @@ public void TestDailyChallengeButton()
             NotificationOverlay notificationOverlay = null!;
             DependencyProvidingContainer buttonContainer = null!;
 
+            AddStep("set intro played flag", () => Dependencies.Get<SessionStatics>().SetValue(Static.DailyChallengeIntroPlayed, true));
```

Comment:
Split this out into an isolated test rather than shoving more and more in such a weird place.

I've started you off in 9b9986b6f2c508c37fb6674c7b47381fa6681148 (my test intentionally fails, you can get it working). Once done you can remove the tests in here as they are not related to menu buttons.

Reply:
I didn't bother splitting the test since all of those reflect on the `DailyChallengeButton` itself, but sure.